benchmarks/src/gatling/scala/org/apache/polaris/benchmarks/actions/AuthenticationActions.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.benchmarks.actions
+
+import io.gatling.core.Predef._
+import io.gatling.core.feeder.Feeder
+import io.gatling.core.structure.ChainBuilder
+import io.gatling.http.Predef._
+import org.apache.polaris.benchmarks.RetryOnHttpCodes.{
+  retryOnHttpStatus,
+  HttpRequestBuilderWithStatusSave
+}
+import org.apache.polaris.benchmarks.parameters.ConnectionParameters
+import org.slf4j.LoggerFactory
+
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Actions for performance testing authentication operations. This class provides methods to
+ * authenticate and manage access tokens for API requests.
+ *
+ * @param cp Connection parameters containing client credentials
+ * @param accessToken Reference to the authentication token shared across actions
+ * @param maxRetries Maximum number of retry attempts for failed operations
+ * @param retryableHttpCodes HTTP status codes that should trigger a retry
+ */
+case class AuthenticationActions(
```

Comment:
Is this part of the benchmarks listed in the config? I don't remember seeing a doc in the readme for this

Reply:
I assume you are talking about the `*Actions.scala` classes.  These classes regroup the elements that are needed to perform certain actions based on either a feature (e.g. authenticating) or an entity (e.g. tables).

They are not listed in the README, indeed, as they are an implementation detail.  This is also where the configuration is consumed, not defined.

Reply:
I mean specifically the Authentication Actions, I didn't understand where that fit in the Write/Read benchmark

Reply:
Ok I see.  The explanation is that before any request can be issued, the user must first authenticate against the OAuth endpoint.  This class takes care of that.  It can feel a little bit over-engineered as, at this time, authentication is performed only once at the very beginning of the benchmark.  Example from the `CreateTreeDatasetConcurrent` simulation.

```
    authenticate
      .inject(atOnceUsers(1))
      .andThen(createCatalogs.inject(atOnceUsers(50)))
      .andThen(
        ..
      )
```

But the OAuth token is only valid for 1h.  Which means that any benchmark that lasts more than that will fail.  I have separate modifications (not published yet) that renew this token periodically that make use of the `AuthenticationActions` more.  Does that help clarify?

Reply:
I believe we should be using the credential refresh hooks now? I haven't followed this enough in the OSS Iceberg a discussion but there should be auto-refresh hooks now but that may just be for the Iceberg client?

Reply:
We could.  But I am not sure what the value would be for OSS Polaris.  I need to research that more.  AFAICT, the OAuth endpoint is deprecated for removal.  Which means there could be a way to bypass authentication entirely later for internal testing purposes.  Alternatively, if it only stays for development purposes, then we don't need to strictly follow the client credentials flow?

Reply:
At some point Polaris will support external IdP (AFAIK), so auto-refresh hooks may not cover all cases.